libs/libcommon/cmake/find_jemalloc.cmake

```
@@ -12,7 +12,15 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-option (ENABLE_JEMALLOC "Set to TRUE to use jemalloc" ON)
+if(${CMAKE_SYSTEM_NAME} MATCHES "Linux")
+    set(ENABLE_JEMALLOC_DEFAULT 1)
+elseif(${CMAKE_SYSTEM_NAME} MATCHES "FreeBSD")
+    set(ENABLE_JEMALLOC_DEFAULT 0)
+elseif(${CMAKE_SYSTEM_NAME} MATCHES "Darwin")
+    set(ENABLE_JEMALLOC_DEFAULT 0)
+endif()
+
+option (ENABLE_JEMALLOC "Set to TRUE to use jemalloc" ${ENABLE_JEMALLOC_DEFAULT})
```

```suggestion
option (ENABLE_JEMALLOC "Set to TRUE to use jemalloc" ON)
```
the jemalloc build issue under macos has been fixed in the master branch since https://github.com/pingcap/tiflash/pull/8847